osu.Game/Overlays/Settings/Sections/SkinSection.cs

```diff
@@ -91,7 +91,7 @@ private void load(OsuConfigManager config)
             if (skinDropdown.Items.All(s => s.ID != configBindable.Value))
                 configBindable.Value = 0;
 
-            configBindable.BindValueChanged(id => dropdownBindable.Value = skinDropdown.Items.Single(s => s.ID == id.NewValue), true);
+            configBindable.BindValueChanged(id => Scheduler.AddOnce(updateSelectedSkinFromConfig), true);
```

Comment:
So my main concern about this is: as far as I understand, this will work as long as `updateSelectedSkinFromConfig` is always scheduled after `itemUpdated`'s operation is scheduled, which I don't think is necessarily guaranteed? Theoretical scenario in which it wouldn't be the case:

```
[DB import thread]				[update thread]

MutableDatabaseBackedStore.Add()
	[adds skin, commits change]

								SkinManager.SelectRandomSkin()
									[selects newly imported and committed skin]
									CurrentSkinInfo.Value is set
									=> OsuGame updates value in config
									=> SkinSection schedules updateSelectedSkinFromConfig()

	ItemUpdated() is raised
	=> SkinSection schedules new dropdown item addition

---- [FRAME N+1] ---

								scheduled updateSelectedSkinFromConfig() runs
									crashes because the item addition is next in schedule queue
```

I don't know how theoretical this is but I don't feel entirely safe with this solution due to relying on schedule order. I'd probably prefer to have guarantees that this can't happen, by having the random skin selection process only select from skins that are known to have had `ItemUpdated()` fired for them - unless that guarantee is already there and I'm not seeing how.

Reply:
I think it's a valid concern, but also a *really* annoying one to fix based on how things are currently flowing. I attempted to clean this up using a `Cached` and invalidation, but this is another case where threading across multiple contexts/instances makes that ugly and unwanted.

Thinking the best solution may be to just add the missing item to the dropdown if it's not already there; how would that sit with you?

Reply:
Sounds like a simple enough stop-gap solution. Would probably leave a comment why it's doing that, but yeah.

Reply:
Have made the change. For testing, you can comment out the code in `itemUpdated` then import some skins and try random selection.